compiler/resolution/callDestructors.cpp

```diff
@@ -1710,17 +1735,53 @@ static void checkForInvalidGlobalUses() {
 //
 // This function simply checks that no function marked with that
 // flag is ever called and raises an error if so.
+//
+// Additionally, also collect in formals that are marked with
+// "error on copy", and check their actuals to see if they are
+// initialized with calls to initCopy().
 static void checkForErroneousInitCopies() {
 
   std::map<FnSymbol*, const char*> errors;
+  std::set<ArgSymbol*> errorOnCopyFormals;
 
-  // Store errors in local map
   forv_Vec(FnSymbol, fn, gFnSymbols) {
     if (fn->hasFlag(FLAG_ERRONEOUS_COPY)) {
       // Store the error in the local map
       if (const char* err = getErroneousCopyError(fn))
         errors[fn] = err;
     }
+
+    // Collect in intent formals that are marked "error on copy".
+    for_formals(formal, fn) {
+      if (formal->hasFlag(FLAG_ERROR_ON_COPY)) {
+        if (formal->intent & INTENT_IN ||
+            formal->originalIntent & INTENT_IN) {
+          errorOnCopyFormals.insert(formal);
+        } else {
+          INT_FATAL(formal, "is marked with %s but is not %s",
+                            "error on copy",
+                            intentDescrString(INTENT_IN));
+        }
+      }
+    }
+  }
+
+  // Iterate through "error on copy" formals and check their callsites.
+  // Emit an error if an actual is a copy.
+  for (ArgSymbol* formal : errorOnCopyFormals) {
```

Comment:
I am hoping this use of C++11 does not cause problems in testing.

Reply:
I can revert for now? I asked Vass and he pointed me to some `auto` filled code on a branch he is working on. I couldn't contain myself.

Reply:
No strong opinion here it's just if it were me I wouldn't want to be the first one to commit code using these features :) We already have override but that's probably more likely to work with old compilers. It is up to you.

Reply:
> I wouldn't want to be the first one to commit code using these features 

At least not the week before feature freeze.

Reply:
I'll use `for_set` for now.